plugins/Devices/CDRom/CDView.vala

```diff
@@ -28,115 +28,115 @@
+
         build_ui ();
-        
+
         dev.initialized.connect (cd_initialised);
-        
+
     }
-    
+
     public void build_ui () {
-        
+
         main_event_box = new Gtk.EventBox ();
         main_grid = new Gtk.Grid ();
-        
+
         /* Content view styling */
         main_event_box.get_style_context ().add_class (Granite.StyleClass.CONTENT_VIEW);
-        
+
         if ( cd_playlist.is_empty () == false) {
         }
-        
+
         album_image = new Widgets.AlbumImage ();
         album_image.gicon = new ThemedIcon ("albumart");
         album_image.halign = Gtk.Align.CENTER;
         album_image.valign = Gtk.Align.CENTER;
         album_image.set_alignment(0.5f, 1);
-        
+
         title = new Gtk.Label ("");
         title.set_alignment(0.5f, 1);
         title.set_justify(Gtk.Justification.CENTER);
         Granite.Widgets.Utils.apply_text_style_to_label (Granite.TextStyle.H2, title);
-        
+
         author = new Gtk.Label ("");
         author.set_alignment(0.5f, 1);
         author.set_justify(Gtk.Justification.CENTER);
         author.sensitive = false;
         Granite.Widgets.Utils.apply_text_style_to_label (Granite.TextStyle.H2, author);
-        
+
         var fake_label_1 = new Gtk.Label ("");
         fake_label_1.set_hexpand (true);
         fake_label_1.set_vexpand (true);
-        
+
         var fake_label_2 = new Gtk.Label ("");
         fake_label_2.set_hexpand (true);
         fake_label_2.set_vexpand (true);
-        
+
         var fake_label_3 = new Gtk.Label ("");
         fake_label_3.set_hexpand (true);
-        
+
         var import_grid = new Gtk.Grid ();
         var import_button = new Gtk.Button.with_label (_("Import"));
         import_button.set_alignment(1, 0);
         import_grid.attach (fake_label_3,  0, 0, 1, 1);
         import_grid.attach (import_button, 1, 0, 1, 1);
-        
+
         main_grid.attach (fake_label_1,   0, 0, 1, 7);
         main_grid.attach (album_image,    1, 3, 1, 1);
         main_grid.attach (title,          2, 2, 1, 1);
         main_grid.attach (author,         3, 2, 1, 1);
         main_grid.attach (cd_viewwrapper, 2, 3, 2, 1);
         main_grid.attach (import_grid,    3, 4, 1, 1);
         main_grid.attach (fake_label_2,   4, 0, 1, 7);
-        
+
         main_event_box.add (main_grid);
         this.attach (main_event_box,0,0,1,1);
-        
+
         /* Create options */
-        
+
         main_grid.set_hexpand (true);
         main_grid.set_row_spacing (6);
         main_grid.set_column_spacing (12);
         main_grid.set_margin_top (12);
-        
+
         import_button.clicked.connect ( () => {dev.transfer_to_library (cd_playlist.medias);});
-        
+
         show_all ();
     }
-    
+
     public void cd_initialised () {
         cd_playlist.add_medias (dev.get_medias ());
         if ( cd_playlist.is_empty () == false) {
-            var m = cd_playlist.medias.peek ();
+            var m = cd_playlist.medias[0];
```

Comment:
You have introduced the `get` and `set` methods in the `Noise.Playlist` class, yet you are referencing `medias` directly, shouldn't `medias` be turned into a private attribute?

Reply:
Yes, I think I'll have to introduce a few more methods to `Playlist`s (at least one to get `media.size`), but it would be better.